contrib/opentimelineio_contrib/application_plugins/rv/example_otio_writer/otio_writer.py

```diff
@@ -0,0 +1,372 @@
+            metadata=get_node_otio_metadata(node_name)
+        )
+    else:
+        # Create the available range
+        start_time, duration = frames_to_rational_times(
+            start_frame=get_movie_first_frame(node_name),
+            end_frame=get_movie_last_frame(node_name),
+            fps=get_movie_fps(node_name)
+        )
+        available_range = otio.opentime.TimeRange(
+            start_time=start_time,
+            duration=duration
+        )
+
+        media_reference = otio.schema.ExternalReference(
+            target_url=("file://" + source_path
+                        if not source_path.startswith("file://")
+                        else source_path),
+            available_range=available_range,
+            metadata=get_node_otio_metadata(source_node)
+        )
+        media_reference.name = get_node_ui_name(node_name)
+        metadata = get_node_otio_metadata(node_name) or {}
+        metadata["RVSourceGroup"] = node_name
+        return otio.schema.Clip(
+            name=get_node_ui_name(node_name),
+            media_reference=media_reference,
+            source_range=source_range,
+            metadata=metadata
+        )
+
+
+def _create_transition(node_name, in_frame=None, out_frame=None,
+                       cut_in_frame=None):
+    """
+    Create an OTIO Transition for a CrossDissolve node.
+    :param node_name: `str`
+    :return: `otio.schema.Transition`
+    """
+    transition_type = TRANSITION_TYPE_MAP.get(
+        rv.commands.nodeType(node_name),
+        otio.schema.transition.TransitionTypes.Custom
+    )
+    if not transition_type:
+        return None
+
+    input_node_names, _ = rv.commands.nodeConnections(node_name)
+
+    # Assume FPS matches first input (RV Transition don't have FPS settings)
+    fps = get_source_fps(input_node_names[0])
+
+    durations_frames = rv.commands.getFloatProperty(
+        node_name + ".parameters.numFrames")
+    if not durations_frames:
+        duration_frames = 20.0
+    else:
+        duration_frames = durations_frames[0]
+
+    in_offset_frames = int(math.floor(duration_frames/2))
+    in_offset = otio.opentime.RationalTime(in_offset_frames, rate=fps)
+    out_offset = otio.opentime.RationalTime(duration_frames-in_offset_frames,
+                                            rate=fps)
+
+    transition = otio.schema.Transition(
+        name=get_node_ui_name(node_name),
+        transition_type=transition_type,
+        in_offset=in_offset,
+        out_offset=out_offset,
+        metadata=get_node_otio_metadata(node_name)
+    )
+    return transition
+
+
+def get_node_ui_name(node_name):
+    """
+    Retrieve the value of the UI Name property for a node, if it exists.
+    :param node_name: `str`
+    :return: `str`
+    """
+    prop_name = node_name + ".ui.name"
+    if rv.commands.propertyExists(prop_name):
+        return rv.commands.getStringProperty(prop_name)[0]
+
+    return None
+
+
+def get_node_otio_metadata(node_name):
```

Comment:
It might be nice to grab all the properties of a node as well.  If the user doesn't follow this metadata scheme using the reader or writing it interactively, there will be data loss.

Reply:
It strikes me as overkill for an example plugin, but lemme play with it a bit.